### PR TITLE
70863 organisation type

### PR DESCRIPTION
**Pull request by seanciv**
https://dev.azure.com/civica-digital/EA%20-%20WEEE/_workitems/edit/70863

This also includes subsequent screens which are just placeholders/ copies of the process here: /admin/add-organisation/type?searchedText=dd&entityType=Aatf

They will need to modified when the work items are picked up

### CODE REVIEW

File: src/EA.Weee.Web/Views/OrganisationRegistration/PartnershipDetails.cshtml [hunk 1]
```diff
@@ -0,0 +1,45 @@
+﻿@using EA.Weee.Core.DataStandards
+@using EA.Weee.Web.Areas.Admin.Controllers
+@using MvcNavigationHelpers
+@using EA.Weee.Web.Controllers
+@model EA.Weee.Web.Areas.Admin.ViewModels.AddOrganisation.Details.PartnershipDetailsViewModel
+@{
+    ViewBag.Title = "Partnership details";
+}
+@section main_content
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/OrganisationRegistration/PartnershipDetails.cshtml:9]: There is a new section full_main_content that provides a wider screen. Think we use that for these new screens.

File: src/EA.Weee.Web/Views/OrganisationRegistration/SoleTraderDetails.cshtml [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+
+        <div class="govuk-form-group @Html.Gds().FormGroupClass(m => m.Address)">
+            @Html.EditorFor(m => m.Address)
+        </div>
+
+        <div class="govuk-form-group">
+            <button class="govuk-button" type="submit" data-module="govuk-button" data-prevent-double-click="true">Continue</button>
+        </div>
+        <p>
+            <a href="@(Url.UrlFor<OrganisationRegistrationController>(u => u.Type(Model.BusinessTradingName)))">Back</a>
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/OrganisationRegistration/SoleTraderDetails.cshtml:48]: There is a BackLink helper now for example

` @(this.WeeeGds().BackLink(@Url.UrlFor<HomeController>(a => a.ChooseActivity(Model.OrganisationId))))`

It should probably be at the top...same think for the other views

File: src/EA.Weee.Web/Controllers/OrganisationRegistrationController.cs [hunk 3]
```diff
@@ -229,6 +234,88 @@ public async Task<ViewResult> JoinOrganisationConfirmation(Guid organisationId,
             }
         }
 
+        [HttpGet]
+        public ActionResult Type(string searchedText)
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Controllers/OrganisationRegistrationController.cs:238]: I don't think we need the search text in this one as we will go back to a different screen. Probably need to adjust this over the coming days as the screen order might change, so for now probably can just remove the parameter.

File: src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs [hunk 4]
```diff
@@ -558,6 +560,85 @@ public async Task PostSearchResults_WithSelectedOrganisationId_RedirectsToJoinOr
...
+            const string organisationType = "Sole trader";
+            var entityType = fixture.Create<EntityType>();
+
+            var result = await controller.SoleTraderDetails(organisationType, searchText) as ViewResult;
+
+            var resultViewModel = result.Model as SoleTraderDetailsViewModel;
+
+            Assert.Equal(searchText, resultViewModel.CompanyName);
+            Assert.Equal(organisationType, resultViewModel.OrganisationType);
+            //Assert.Equal(countries, resultViewModel.Address.Countries);
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web.Tests.Unit/Controllers/OrganisationRegistrationControllerTests.cs:637]: Should this be removed?

File: src/EA.Weee.Web/Views/OrganisationRegistration/PartnershipDetails.cshtml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+        </h1>
+    </header>
+
+    @using (Html.BeginForm())
+    {
+        @Html.AntiForgeryToken()
+        @Html.Gds().ValidationSummary()
+        @Html.HiddenFor(m => m.OrganisationType)
+
+        <div class="govuk-form-group @Html.Gds().FormGroupClass(m => m.BusinessTradingName)">
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/OrganisationRegistration/PartnershipDetails.cshtml:23]: May need to adjust some of these fields but will see what Dave comes back with.

File: src/EA.Weee.Web/Views/OrganisationRegistration/Type.cshtml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+        @this.WeeeGds().RadioButtonsFor(
+            m => m.SelectedValue,
+            Model.PossibleValues,
+            "Organisation type",
+            RadioButtonLegend.VisuallyHidden,
+            RadioButtonLayout.Stacked)
+    </div>
+
+    <div class="govuk-form-group">
+        <button class="govuk-button" type="submit" data-module="govuk-button" data-prevent-double-click="true">Continue</button>
```
Comment (JohnPadgettCivica) [inline src/EA.Weee.Web/Views/OrganisationRegistration/Type.cshtml:41]: You can use

` @(this.WeeeGds().Submit("Continue"))`

**Review by JohnPadgettCivica — COMMENTED** [review]
Some minor comments

**Comment by seanciv** [comment]
I have dealt with the Type page comments and will fix the other ones when those pages are picked up.

**Pull request merged**